actions/views/rhs.js

```diff
@@ -104,11 +104,13 @@ export function getFlaggedPosts() {
         await PostActions.getProfilesAndStatusesForPosts(result.posts, dispatch, getState);
 
         dispatch(receivedSearchPosts(teamId, result));
+        dispatch({type: ActionTypes.SEARCH_FLAGGED_POSTS_SUCCESS});
```

Comment:
I was going to ask you to [batch these actions](https://github.com/mattermost/mattermost-webapp/blob/master/actions/views/rhs.js#L77) because if dispatched separately, redux will register two state changes and run all the mapDispatchToProps twice. If dispatched in a batch, only one state change occurs

But `receivedSearchPosts` is already returning a batched action and I don't think it supports batching batches (you can try if you like)

It might be a good improvement ticket to go through all this file's actions and batch the dispatches where possible

Reply:
I didn't know that. Thanks for pointing out.
I'd like to have it on a separate ticket which I filed as [PLT-8449](https://mattermost.atlassian.net/browse/PLT-8449)